crates/recursor/src/recursor.rs

```diff
@@ -638,6 +659,16 @@ mod for_dnssec {
             .boxed()
         }
     }
+
+    pub(super) trait DnssecPoolContext {
+        fn pool_context(&self) -> Arc<PoolContext>;
+    }
+
+    impl<P: ConnectionProvider> DnssecPoolContext for DnssecDnsHandle<RecursorDnsHandle<P>> {
```

Comment:
Not sure why this complexity is needed. If `DnssecDnsHandle` gives access to its inner handle and the inner handle gives access to `pool_context()`, why do we need an extension trait here?

Reply:
Oh!!! I was missing the fact that `RecursorMode<P>::Validating`'s handle isn't a generic `DnssecDnsHandle<H>` it's specifically a `DnssecDnsHandle<RecursorDnsHandle<P>>` (obviously...).

Agreed that the backflips I was doing here are completely unnecessary then. :face_exhaling: